### PR TITLE
Have go-to-def flip between partial definition and implementation

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/76156

### CODE REVIEW

File: src/EditorFeatures/Core/Navigation/AbstractDefinitionLocationService.cs [hunk 1]
```diff
@@ -251,6 +249,24 @@ async ValueTask<ImmutableArray<DefinitionItem>> GetInterceptorDefinitionsAsync(
                 });
             }
         }
+        async ValueTask<INavigableLocation?> TryGetOtherPartOfPartialAsync()
```
Comment (ToddGrun) [inline src/EditorFeatures/Core/Navigation/AbstractDefinitionLocationService.cs:252]: >async ValueTask<INavigableLocation?> TryGetOtherPartOfPartialAsync() [](http://example.com/codeflow?start=8&length=68)

nit: missing newline

**Review by ToddGrun — APPROVED** [review]
:shipit:

File: src/EditorFeatures/Core/Navigation/AbstractDefinitionLocationService.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/Navigation/AbstractDefinitionLocationService.cs:252]: Will fix in followup.

**Comment by Shujee** [comment]
Just trying to understand the status. Does this fix the problem where F12-ing a generated `ObservableProperty` takes us to `.g.cs` file instead of actual underlying local field? If so, has this been pushed to the main updates channel? When can we expect it to appear in GA version of VS2022?

**Comment by CyrusNajmabadi** [comment]
> Just trying to understand the status. Does this fix the problem where F12-ing a generated `ObservableProperty` takes us to `.g.cs` file instead of actual underlying local field? 

F12 on a partial will take you to one of the parts of the partial.  F12 on that partial will take you to the other part (and vice versa). So you can trivially navigate to whichever you prefer. 

> If so, has this been pushed to the main updates channel? When can we expect it to appear in GA version of VS2022

 A bot adds that information when this merges into the main vs tree.

**Pull request merged**